### PR TITLE
[Dynamo] No graph break for super(MyConv{1/2/3}d, self).forward and super(MyConvTranspose, self).forward

### LINKED ISSUE #101155: [Dynamo] Graph break: call_method NNModuleVariable() _conv_forward
### 🐛 Describe the bug

Repro:
```
import torch
import logging
import torch._dynamo

# torch._logging.set_logs(dynamo=logging.DEBUG, bytecode=True)
torch._dynamo.config.print_graph_breaks = True

import torch.nn as nn
import torch.nn.functional as F

class MyConv1d(nn.Conv1d):

    def __init__(self, in_channels, out_channels, kernel_size, **kwargs):
        super(MyConv1d, self).__init__(
            in_channels=in_channels,
            out_channels=out_channels,
            kernel_size=kernel_size,
            **kwargs
        )

    def forward(self, inputs, mask=None):
        outputs = super(MyConv1d, self).forward(inputs)
        return outputs

x = torch.randn(4, 4)

m = MyConv1d(4, 4, 4)
print(m(x))

opt_m = torch.compile(backend="eager")(m)
print(opt_m(x))
```

Graph breaks:
```
[2023-05-11 05:08:54,218] torch._dynamo.symbolic_convert: [WARNING] Graph break: call_method NNModuleVariable() _conv_forward [TensorVariable(), TensorVariable(), TensorVariable()] {} from user code at   File "/scratch/ybliang/work/repos/debug/debug3.py", line 22, in forward
    outputs = super(MyConv1d, self).forward(inputs)
  File "/scratch/ybliang/work/repos/pytorch/torch/nn/modules/conv.py", line 313, in forward
    return self._conv_forward(input, self.weight, self.bias)
```

### Versions

N/A

cc @ezyang @soumith @msaroufim @wconstab @ngimel @bdhirsh @anijain2305 @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### PR DESCRIPTION
**Pull request by weifengpy**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #102509

before the PR, running super(MyConv1d, self).forward or super(MyConvTranspose, self).foward, dynamo will create a graph break when executing NNModuleVariable.call_method and raise unimplemented error for name=_conv_forward / _output_padding. see issue for full detail: https://github.com/pytorch/pytorch/issues/101155

after the PR, for torch.nn.conv module with function name _conv_forward / _output_padding, we inline the function with tx.inline_user_function_return

code refactor: added NNModuleVariable._inline_user_function_return_helper to consolidaste tx.inline_user_function_return into 1 place to keep code dry. after factor, there are 2 uncolidated inline_user_function_return with different ```fn``` and ```source``` logic. the code is still dry. For local testing, they are covered by test_modulelist, test_moduledict, test_conv_call_super_forward_directly and test_conv_transpose_call_super_forward_directly in test_modules.py

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy

Differential Revision: [D46494460](https://our.internmc.facebook.com/intern/diff/D46494460)

### CODE REVIEW

File: torch/_dynamo/variables/nn_module.py [hunk 1]
```diff
@@ -606,6 +606,18 @@ def gen_source(source, name):
                 [self] + args,
                 kwargs,
             )
+        elif name == "_conv_forward":
+            # inline the torch.nn.modules.conv._conv_forward
+            fn = getattr(module, name).__func__
+            module_attr = getattr(fn, "__module__", "")
+            if module_attr is not None and module_attr.startswith(
+                "torch.nn.modules.conv"
+            ):
```
Comment (yanboliang) [inline torch/_dynamo/variables/nn_module.py:615]: I would suggest to use ```isinstance(mod, torch.nn.modules.conv._ConvNd```, and you can get ```mod``` by ```tx.output.get_submodule(self.module_key)```. If users extends from the builtin ```Conv``` module, they can benefit from this inline. We do see similar patterns from torchbench models.

File: torch/_dynamo/variables/nn_module.py [hunk 2]
```diff
@@ -606,6 +606,18 @@ def gen_source(source, name):
...
+                return tx.inline_user_function_return(
```
Comment (yanboliang) [inline torch/_dynamo/variables/nn_module.py:616]: There are many places finally navigate to ```inline_user_function_return```, can you try to consolidate them?

**Review by yanboliang — COMMENTED** [review]
This looks correct overall, I left two small comments. BTW, can you also add supports/tests for ```_ConvTransposeNd. _output_padding```? Thank you!

File: torch/_dynamo/variables/nn_module.py [hunk 3]
```diff
@@ -599,12 +639,44 @@ def gen_source(source, name):
...
+        ):
+            # inline the torch.nn.modules.conv._ConvTransposeNd._output_padding
+            return self._inline_user_function_return_helper(
+                tx,
+                module,
+                name,
+                args,
                 kwargs,
+                options,
+                function_type=InlineFunctionType.METHOD,
```
Comment (yanboliang) [inline torch/_dynamo/variables/nn_module.py:679]: I believe we can consolidate all of these into:
```
elif name == "_get_abs_string_index" or (
    isinstance(module, torch.nn.modules.conv._ConvNd)
    and name == "_conv_forward"
) or (
    isinstance(module, torch.nn.modules.conv._ConvTransposeNd)
    and name == "_output_padding"
):
    # Inline the function
    fn = getattr(module, name).__func__
    fn_source = AttrSource(self.source, "__func__")
    options["source"] = fn_source
    return tx.inline_user_function_return(
        variables.UserFunctionVariable(fn, **options),
        [self] + args,
        kwargs,
    )
```

Comment (yanboliang) [reply torch/_dynamo/variables/nn_module.py:679]: We can use either ```UserFunctionVariable``` or ```UserMethodVariable``` here, just need to manipulate args to what they expect and update source correctly.

Comment (yanboliang) [reply torch/_dynamo/variables/nn_module.py:679]: In the future, we can use this function as the default handling logic for ```NNModuleVariable.call_method```.

**Review by yanboliang — APPROVED** [review]

**Comment by willfengg** [comment]
@willfengg has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D46494460).

**Pull request closed without merging**